modules/generative-anthropic/clients/anthropic.go

```diff
@@ -0,0 +1,275 @@
+		}
+
+		var resBody anthropicApiError
+		if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+			return nil, errors.Wrap(err, "unmarshal response body")
+		}
+
+		return nil, fmt.Errorf("Anthropic API error: %s - %s", resBody.Error.Type, resBody.Error.Message)
+
+	}
+
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "read response body")
+	}
+
+	var resBody generateResponse
+	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	textResponse := resBody.Content[0].Text
+	if len(resBody.Content) > 0 && textResponse != "" {
+		trimmedResponse := strings.Trim(textResponse, "\n")
+		return &generativemodels.GenerateResponse{
+			Result: &trimmedResponse,
+		}, nil
+	}
+
+	return nil, errors.New("no response from Anthropic")
+}
+
+func (a *anthropic) getAnthropicUrl(ctx context.Context, baseURL string) (string, error) {
+	passedBaseURL := baseURL
+	if headerBaseURL := a.getValueFromContext(ctx, "X-Anthropic-Baseurl"); headerBaseURL != "" {
+		passedBaseURL = headerBaseURL
+	}
+	return url.JoinPath(passedBaseURL, "/v1/messages")
+}
+
+func (a *anthropic) generatePromptForTask(textProperties []map[string]string, task string) (string, error) {
+	marshal, err := json.Marshal(textProperties)
+	if err != nil {
+		return "", errors.Wrap(err, "marshal text properties")
+	}
+	task = compile.ReplaceAllStringFunc(task, func(match string) string {
+		match = strings.Trim(match, "{}")
+		for _, textProperty := range textProperties {
+			if val, ok := textProperty[match]; ok {
+				return val
+			}
+		}
+		return match
+	})
+	return fmt.Sprintf(task, marshal), nil
+}
+
+func (a *anthropic) generateForPrompt(textProperties map[string]string, prompt string) (string, error) {
+	all := compile.FindAll([]byte(prompt), -1)
+	for _, match := range all {
+		originalProperty := string(match)
+		replacedProperty := compile.FindStringSubmatch(originalProperty)[1]
+		replacedProperty = strings.TrimSpace(replacedProperty)
+		value := textProperties[replacedProperty]
+		if value == "" {
+			return "", errors.Errorf("Following property has empty value: '%v'. Make sure you spell the property name correctly, verify that the property exists and has a value", replacedProperty)
+		}
+		prompt = strings.ReplaceAll(prompt, originalProperty, value)
+	}
+	return prompt, nil
+}
+
+func (a *anthropic) getValueFromContext(ctx context.Context, key string) string {
+	if value := ctx.Value(key); value != nil {
+		if keyHeader, ok := value.([]string); ok && len(keyHeader) > 0 && len(keyHeader[0]) > 0 {
+			return keyHeader[0]
+		}
+	}
+	// try getting header from GRPC if not successful
+	if apiKey := modulecomponents.GetValueFromGRPC(ctx, key); len(apiKey) > 0 && len(apiKey[0]) > 0 {
+		return apiKey[0]
+	}
+	return ""
+}
+
+func (a *anthropic) getApiKey(ctx context.Context) (string, error) {
+	if apiKey := a.getValueFromContext(ctx, "X-Anthropic-Api-Key"); apiKey != "" {
```

Comment:
please add header value to allowed header [list](https://github.com/weaviate/weaviate/blob/main/usecases/config/config_handler.go#L291)